### PR TITLE
[libc][type_traits] Add aligned_storage

### PR DESCRIPTION
**Pull request by PiJoules**

### CODE REVIEW

**Comment by michaelrj-google** [comment]
This patch has reminded me that I need to simplify the bazel a bit, so definitely don't land it before this patch lands: https://github.com/llvm/llvm-project/pull/94079

**Review by lntue — APPROVED** [review]

**Review by lntue — CHANGES_REQUESTED** [review]
You'll need to add the new header to the `type_traits` target in `src/__support/CPP/CMakeLists.txt`

**Comment by PiJoules** [comment]
> You'll need to add the new header to the `type_traits` target in `src/__support/CPP/CMakeLists.txt`

Done

**Review by lntue — APPROVED** [review]

**Pull request merged**
